panic with new key-value logic

I received the following panic with go-fuzz a60c04bd38158403027a022ce604eed9cbe74d60
while fuzzing github.com/Sereal/Sereal/Go/sereal.

Looks like the new key-value stuff that was just merged.

```
panic: runtime error: slice bounds out of range

goroutine 50 [running]:
github.com/dvyukov/go-fuzz/go-fuzz/versifier.structureKeyValue(0xc82027bb00, 0x6, 0x8, 0x0, 0x0, 0x0)
        /home/dgryski/work/src/gocode/src/github.com/dvyukov/go-fuzz/go-fuzz/versifier/versifier.go:696 +0x75e
github.com/dvyukov/go-fuzz/go-fuzz/versifier.structure(0xc82027bb00, 0x6, 0x8, 0x0, 0x0, 0x0)
        /home/dgryski/work/src/gocode/src/github.com/dvyukov/go-fuzz/go-fuzz/versifier/versifier.go:575 +0xdd
github.com/dvyukov/go-fuzz/go-fuzz/versifier.BuildVerse(0x0, 0xc8202e0330, 0xf, 0x2b, 0x10000)
        /home/dgryski/work/src/gocode/src/github.com/dvyukov/go-fuzz/go-fuzz/versifier/versifier.go:55 +0x1b9
main.(*Hub).loop(0xc820200e00)
        /home/dgryski/work/src/gocode/src/github.com/dvyukov/go-fuzz/go-fuzz/hub.go:225 +0x19ed
created by main.newHub
        /home/dgryski/work/src/gocode/src/github.com/dvyukov/go-fuzz/go-fuzz/hub.go:124 +0x10e7
```

--- a/go-fuzz/versifier/versifier.go
+++ b/go-fuzz/versifier/versifier.go
@@ -692,8 +692,9 @@ func structureKeyValue(nn []Node) (res []Node) {
 					continue
 				}
 			}
-			kv := &KeyValNode{ctrl.ch, key, value}
-			nn = append(nn[:i-1], append([]Node{kv}, nn[i+2:]...)...)
+			nn[i+1] = &KeyValNode{ctrl.ch, key, value}
+			copy(nn[i-1:], nn[i+1:])
+			nn = nn[:len(nn)-2]
 		}
 	}
 	return nn

